src/main/java/pl/craftserve/radiation/BaseConfig.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 Aleksander Jagiełło <themolkapl@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pl.craftserve.radiation;
+
+import org.bukkit.ChatColor;
+import org.bukkit.Server;
+import org.bukkit.boss.BarColor;
+import org.bukkit.boss.BarFlag;
+import org.bukkit.boss.BarStyle;
+import org.bukkit.boss.BossBar;
+import org.bukkit.configuration.ConfigurationSection;
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.MemoryConfiguration;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class BaseConfig {
+    private static final char COLOR_CODE = '&';
+
+    protected String colorize(String input) {
+        return input == null ? null : ChatColor.translateAlternateColorCodes(COLOR_CODE, input);
+    }
+
+    public interface BarConfig {
+        String title();
+        BarColor color();
+        BarStyle style();
+        BarFlag[] flags();
+
+        default BossBar create(Server server) {
+            Objects.requireNonNull(server, "server");
+            String title = Objects.toString(this.title(), "");
+            return server.createBossBar(title, this.color(), this.style(), this.flags());
+        }
+    }
+
+    public static class BarConfigImpl extends BaseConfig implements BarConfig {
```

Comment:
Po co tak wlasciwie ta klasa dziedziczy z BaseConfig i jeszcze te interfejsy? Komplikujesz ;)

Reply:
Tam jest kod wspólny, np metoda `colorize` odpowiadająca za kolorowanie tekstu.

Reply:
Ktora ma jedna linijke, lepiej skopiowac ;)

Reply:
Nie lepiej bo to wtedy nie jest DRY. Tam potrzebny jest color code, lepiej to mieć w jednym miejscu.